### ISSUE TITLE
Fragments containing multiple blocks within matrix fields do not return their subsets

### ISSUE THREAD

**Issue by toddpadwick** [craftcms/cms#9788]
### Description
Here is an example matrix field fragment:
```
    fragment landingPageBuilderFrag on landingPageBuilder_MatrixField {
      ... on landingPageBuilder_sectionHeadline_BlockType {
        typeHandle
        sectionTitle
        headline
        introduction
      }
      ... on landingPageBuilder_logoParade_BlockType {
        typeHandle
        organisations {
          url
          title
          slug
          id
          uri
          ... on organisations_organisation_Entry {
            logo {
              extension
              width
              height
              url
              id
              title
              srcset(sizes: ["100", "300", "400"])
            }
          }
        }
      }
      ... on landingPageBuilder_faqs_BlockType {
        typeHandle
        faqs {
			    ... on faqs_BlockType {
            question
            answer
          }
        }
        callToAction {
          ... on callToAction_BlockType {
        	    ... callToActionSuperTableFrag     
          }
        }
      }
    }

```

I would like to then use this fragment for multiple entry types which contain this field, for example:

```
    ... on homepage_homepage_Entry {
          landingPageBuilder {
              ...landingPageBuilderFrag
          }
        }

```

In the GraphiQL explorer, this works fine. But when queried via our Nuxt Apollo app it returns each matrix field but without their contents. here is a screenshot of the response in the console:

![Screenshot 2021-09-02 at 17 08 23](https://user-images.githubusercontent.com/36191462/131868996-b6608ba9-c099-497b-9639-4a382fef1d19.png)

as you can see non of the subset data is returned for any of the matrix blocks. 
It looks like someone had a similiar issue in [this ticket](https://github.com/craftcms/cms/issues/5031) but it has been closed and solved. Is it related? or is something else?

### Additional info

- PHP version: 8.0.2
- OS version: Linux 4.15.0-50-generic
- Database driver & version: MySQL 8.0.23
- Craft edition & version: Craft Pro 3.7.10

**Comment by andris-sevcenko** [comment]
As a rule of thumb, if the query works in Craft CMS' GraphiQL IDE, it's not going to be Craft's fault.

I'm not very familiar with Nuxt/Apollo apps, so any information is appreciated:

* Is the correct token used for the queries?
* Are the queries executed on-demand or is there some pre-fetching happening? For example, if you're building a Gatsby JS site and using our Gatsby Source plugin, all the data is fetched into Gatsby data stores beforehand and then GraphQL queries are executed against Gatsby datastore. I'm not implying anything here, just illustrating how query flows can be quite different for different apps and that can influence where to look for the problem.
* If you query for some other fields besides `landingPageBuilder`, are you getting the content returned there? If yes, that would imply something going wrong with matrix field queries. If not, then that would imply query being blocked somewhere altogether.
* If the queries via Nuxt/Apollo end up being executed directly against Craft as they are executed in the app, it's probably worth tapping into the PHP code and add some debugging code to see what the incoming query is and what the response is.

**Comment by toddpadwick** [comment]
Hi @andris-sevcenko  
Thanks for responding so fast...
To answer your questions:
1.  yes its correct
2. I have turned off prefetch to test and i get the same problem
3. all other fields work fine - its just nested fields that don't such as matrix and supertables - i can also confirm they work if i do a fragment for each individual matrix block. which obviously is not ideal as a site grows.
4. I'll have a go at this. as you say, maybe its an issue with Nuxt's Apollo module.

**Comment by andris-sevcenko** [comment]
Perfect, let me know!

**Comment by toddpadwick** [comment]
I've found the fix. it was my fault:

My Nuxt set-up generates the schema for the GraphQL Introspective Fragment matcher. But the schema was no longer updated to the latest due to an error in my config caused by a change i accidentally made a few weeks back and so it had the old schema from the start without all the new data.

If anyone else stumbles upon this issue and is having similar problems I gave a detailed answer on how to get fragments working with Nuxt, Apollo and Craft a while back on this [StackOverflow thread ](https://stackoverflow.com/a/66066033/9067927). But when Nuxt releases v3, I'm told we'll no longer need any of that as it'll work straight out of the box hopefully.
